### PR TITLE
Update text button styles

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #1187 by @obulat

## Description

This PR updates all of the "text" variant buttons with the variants suggested by @fcoveram in #1187:

1. "Clear" button in the Recent searches popover/modal

> Button should be Small + Transparent Gray

3. "Clear filters" button in the Content settings modal footer

> Button should be Large + Transparent Gray

4. "Clear filters" in VSearchGridFilter

> Button should be Small + Transparent Gray

For the buttons that should have no border and no background color, we can use two styles:
- `transparent-tx` - sets border as transparent by default in VButton. It works well for all the buttons that don't change their border color on hover/focus.
- `plain` - does not set any colors. This makes it easier to use when a button needs to change colors on hover. This is more difficult if we set the border color as in `transparent-tx` because we would have to override it in all states with `!` important.

## Testing Instructions

The CI should pass and there should be no problems in code. 
These buttons previously did not have a border, and now they do (even if transparent), so they became 2px wider - this is why the snapshots changed.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VButton.vue [hunk 1]
```diff
@@ -355,33 +355,17 @@ a.button {
...
-  @apply border-tx bg-tx text-dark-charcoal hover:bg-dark-charcoal-10;
+  @apply border-tx bg-tx text-dark-charcoal hover:bg-dark-charcoal-10 disabled:text-dark-charcoal-40;
 }
 .transparent-dark {
   @apply border-tx bg-tx text-dark-charcoal hover:bg-dark-charcoal hover:text-white;
 }
 
-.secondary-bordered {
-  @apply border-dark-charcoal bg-tx hover:bg-dark-charcoal hover:text-white focus-visible:border-tx disabled:bg-dark-charcoal-10 disabled:text-dark-charcoal-40;
-}
```
Comment (AetherUnbound) [inline frontend/src/components/VButton.vue:367]: Very cool to see some of these get consolidated/removed!

**Review by AetherUnbound — APPROVED** [review]
Ran locally without issues and the buttons look good!

**Review by fcoveram — CHANGES_REQUESTED** [review]
**"Clear" button in the Recent searches popover/modal**
Text style should be `label bold`

**"Clear filters" button in the Content settings modal footer**
Text style should be `label bold`

**"Clear filters" in VSearchGridFilter**
Buttons shows its focus ring in `hover`

**Review by zackkrida — APPROVED** [review]
The finale to such a big undertaking 😄 Nice work; lgtm!

**Pull request merged**